From 3fd45557978593addd7291a221f983805d14a36f Mon Sep 17 00:00:00 2001
From: Tobias <ruck.tobias@gmail.com>
Date: Tue, 13 Apr 2021 09:38:53 -0500
Subject: [PATCH 080/196] Fix: Make getblocktemplate create a standard
 scriptDummy (#146)

Otherwise, getblocktemplate fails with "scriptpubkey", as the dummy output created is non-standard.
---
 src/rpc/mining.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/rpc/mining.cpp b/src/rpc/mining.cpp
index b95ab2d93..f762fcdea 100644
--- a/src/rpc/mining.cpp
+++ b/src/rpc/mining.cpp
@@ -888,8 +888,9 @@ static UniValue getblocktemplate(const Config &config,
         CBlockIndex *pindexPrevNew = ::ChainActive().Tip();
         nStart = GetTime();
 
-        // Create new block
-        CScript scriptDummy = CScript() << OP_TRUE;
+        // Create new block. scriptDummy is just for testing block validity and
+        // is not part of the returned template.
+        CScript scriptDummy = CScript() << OP_RETURN;
         pblocktemplate =
             BlockAssembler(config, mempool).CreateNewBlock(scriptDummy);
         if (!pblocktemplate) {
-- 
2.39.5 (Apple Git-154)

